src/phase0/state_transition/epoch_processing.rs

```diff
@@ -86,6 +90,74 @@ pub fn get_matching_target_attestations<
         .filter(move |a| a.data.target.root == *block_root))
 }
 
+pub fn get_matching_head_attestations<
+    'a,
+    const SLOTS_PER_HISTORICAL_ROOT: usize,
+    const HISTORICAL_ROOTS_LIMIT: usize,
+    const ETH1_DATA_VOTES_BOUND: usize,
+    const VALIDATOR_REGISTRY_LIMIT: usize,
+    const EPOCHS_PER_HISTORICAL_VECTOR: usize,
+    const EPOCHS_PER_SLASHINGS_VECTOR: usize,
+    const MAX_VALIDATORS_PER_COMMITTEE: usize,
+    const PENDING_ATTESTATIONS_BOUND: usize,
+>(
+    state: &'a BeaconState<
+        SLOTS_PER_HISTORICAL_ROOT,
+        HISTORICAL_ROOTS_LIMIT,
+        ETH1_DATA_VOTES_BOUND,
+        VALIDATOR_REGISTRY_LIMIT,
+        EPOCHS_PER_HISTORICAL_VECTOR,
+        EPOCHS_PER_SLASHINGS_VECTOR,
+        MAX_VALIDATORS_PER_COMMITTEE,
+        PENDING_ATTESTATIONS_BOUND,
+    >,
+    epoch: Epoch,
+    context: &Context,
+) -> Result<impl Iterator<Item = &'a PendingAttestation<MAX_VALIDATORS_PER_COMMITTEE>>, Error> {
+    let matching_target_attestations = get_matching_target_attestations(state, epoch, context)?;
+    Ok(matching_target_attestations.filter(|a| {
+        a.data.beacon_block_root == *get_block_root_at_slot(state, a.data.slot).unwrap()
```

Comment:
I had to `unwrap()` here -- just checking if this is the desired way to handle things or not. If not, lmk -- I also do this in `get_inclusion_delay_deltas` in 3 different spots.

My understanding is that I'd have to convert this to a `for` loop (or some other methods like `try_into`) to properly use `?` instead of `unwrap()` due this occurring in a closure.